examples/header-group.html

```diff
@@ -0,0 +1,80 @@
+<template id=header-group>
+
+<style>
+
+  header-group
+    { display: block }
+
+  header-group
+    hgroup
+      { background: #bada55; color: var(--darkest-color) }
+
+</style>
+
+<hgroup>
+  <h1><a href=/examples/index.html>snuggsi ツ</a></h1>
+  <slot name=header><h2>try an example</h2></slot>
+</hgroup>
+
+</template>
+
+<script>
+const tag = `header-group`
+
+Element `header-group`
+
+(class extends HTMLElement {
+
+  static onimportload (event) {
+    let template = event.target.import
+      .querySelector ('template')
+
+      document
+        .querySelector('head')
+        .appendChild (template)
+
+      // possibly should be adopting node into document
+      // and should do it on DOMContentLoaded hook up onimportload
+
+    const shadow = function (element) {
+      let fragment = template.content.cloneNode (true)
+
+      fragment.slots =
+        Array.from (fragment.querySelectorAll ('slot'))
+
+      element.slots =
+        Array.from (element.querySelectorAll ('[slot]'))
+
+      element.slots.map (function (namedslot) {
+        fragment.slots
+          .filter (slot =>
+            (slot.getAttribute ('name') === namedslot.getAttribute ('slot'))
+          )
+          .map (slot => {
+            slot.innerHTML = ''
+            , slot.appendChild (namedslot)
```

Comment:
i don't know how i feel about `<slot>` yet i just stumbled across it on MDN and thought i'd like to see what it does natively. if it does what i was expecting natively, then snuggsi should just preserve that native behavior as closely as possible, I guess